linera-core/src/join_set_ext.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! An extension trait to allow determining at compile time how tasks are spawned on the Tokio
+//! runtime.
+//!
+//! In most cases the [`Future`] task to be spawned should implement [`Send`], but that's
+//! not possible when compiling for `wasm32-unknown-unknown`. In that case, the task is
+//! spawned in a [`LocalSet`][`tokio::tast::LocalSet`].
+
+use std::future::Future;
+
+use tokio::{
+    sync::oneshot,
+    task::{AbortHandle, JoinSet},
+};
+
+/// An extension trait for the [`JoinSet`] type.
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
This should definitely be gated on `cfg(not(web))`, not `cfg(not(target_arch = "wasm32"))` (and likewise elsewhere).

The thing that makes our futures non-`Send` is not that we're on a `wasm32` target — it's that the JavaScript runtime doesn't support sharing objects between threads.

Reply:
Oops, my bad. Fixed, thanks!